### PR TITLE
Incorporate transaction for uploading participants

### PR DESCRIPTION
**Pull request by bgfoshay**
## What’s changing?

Updating Participants upload logic (in ParticipantDao.cs) to wrap participant insertion within a database transaction so that all participants will be committed at the same time. Added a unit test to cover scenario where an exception occurs during upload.

## Why?

Bulk Upload Performance testing found performance issues and we noticed that participants were committed to the database one by one. This change could boost upload performance.

## This PR has:

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

**Comment by kingcomma** [comment]
*Question:* If this transaction fails and no participant records for the upload are inserted, will the upload row still exist in the `uploads` table?

If that is the case, the previous upload would effectively be replaced by an empty upload. I hadn't picked up on this implication until seeing the change put into code. 🤔

**Comment by bgfoshay** [comment]
@kingcomma Placed calls to AddUpload & AddParticipants within a TransactionScope. Let me know what you think.

**Review by kingcomma — APPROVED** [review]
👍 Nice solution! Also really appreciate the comprehensive tests.

Just one chore related to dependency versions that would be great to clear up before merging. I'm assuming it's not a big lift, we can reassess if it is.

File: participants/tests/Piipan.Participants.Core.IntegrationTests/Piipan.Participants.Core.IntegrationTests.csproj [hunk 1]
```diff
@@ -8,10 +8,13 @@
 
   <ItemGroup>
     <PackageReference Include="Dapper" Version="2.0.123" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="6.0.1" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="6.0.0" />
```
[the remark this answers is not in the document]
Comment (bgfoshay) [reply participants/tests/Piipan.Participants.Core.IntegrationTests/Piipan.Participants.Core.IntegrationTests.csproj:12]: @kingcomma Another nice catch. Sorry I missed another detail. I don't expect that change to be a heavy lift. I will make the change and resubmit. Thanks for the feedback!

[the remark this answers is not in the document]
Comment (kingcomma) [reply participants/tests/Piipan.Participants.Core.IntegrationTests/Piipan.Participants.Core.IntegrationTests.csproj:12]: Great, glad to hear it's not too much of a chore! Zero rush, feel free to make the edit when your schedule allows.

[the remark this answers is not in the document]
Comment (bgfoshay) [reply participants/tests/Piipan.Participants.Core.IntegrationTests/Piipan.Participants.Core.IntegrationTests.csproj]: Fixed the dependencies.

**Pull request merged**